src/components/CustomerQuota/ItemsSelection/AddonsItems.tsx

```diff
@@ -1,13 +1,72 @@
-import React, { FunctionComponent } from "react";
-import { ActivityIndicator } from "react-native";
+import React, { FunctionComponent, useContext, useEffect } from "react";
+import { ActivityIndicator, View } from "react-native";
 import { color } from "../../../common/styles";
+import { AlertModalContext } from "../../../context/alert";
+import { AuthContext } from "../../../context/auth";
+import { CampaignConfigContext } from "../../../context/campaignConfig";
+import { usePastTransaction } from "../../../hooks/usePastTransaction/usePastTransaction";
+import { useTranslate } from "../../../hooks/useTranslate/useTranslate";
+import { TransactionsGroup } from "../TransactionsGroup";
+import {
+  groupTransactionsByCategory,
+  sortTransactionsByCategory,
+} from "../utils";
 
-export const AddonsItems: FunctionComponent = () => {
-  return (
+const BIG_NUMBER = 99999;
+
+export const AddonsItems: FunctionComponent<{
+  ids: string[];
+  isShowAddonItems: boolean;
+  categoryFilter?: string[];
+}> = ({ ids, isShowAddonItems, categoryFilter }) => {
+  const { sessionToken, endpoint } = useContext(AuthContext);
+  const { policies: allProducts } = useContext(CampaignConfigContext);
+  const {
+    pastTransactionsResult: sortedTransactions,
+    loading,
+    error,
+  } = usePastTransaction(ids, sessionToken, endpoint, categoryFilter);
+
+  const { showErrorAlert } = useContext(AlertModalContext);
+  useEffect(() => {
+    if (error) {
+      showErrorAlert(error);
+    }
+  }, [error, showErrorAlert]);
+  const latestTransactionTime: Date | undefined =
+    sortedTransactions && sortedTransactions.length > 0
+      ? sortedTransactions[0].transactionTime
+      : undefined;
+
+  const translationProps = useTranslate();
+  const transactionsByCategoryMap = groupTransactionsByCategory(
+    sortedTransactions,
+    allProducts || [],
+    latestTransactionTime,
+    translationProps
+  );
+
+  const transactionsByCategoryList = sortTransactionsByCategory(
+    transactionsByCategoryMap
+  );
+  return loading && isShowAddonItems ? (
```

Comment:
If `loading = false` and `isShowAddonItems = false`, then we'll show the list of transactions. Not sure if this is expected, what do you think?

Reply:
however the `maxTransactionToDisplay` will result in 0, so it will not render any transaction. My idea was so that the transactions are preloaded such that there won't be additional fetch

Reply:
Yep so there are 2 points of logic that affect this conditional render, `isShowAddonItems` and also `maxTransactionsToDisplay`. This can be confusing to reason about. Note that `maxTransactionsToDisplay` is meant as a way to limit the number of transactions on screen, less of a boolean to control if any transactions should be shown.
Also, given that the rendering of `AddonsItems` will already fetch the past transactions, the fetch will always take place.

What if we separate out the logic of whether `loading` and whether `isShowAddonItems` is true? i.e. if loading, regardless of isShowAddonItems, show the activity indicator. If not loading, then check for `isShowAddonItems`